hlink/linking/model_exploration/link_step_train_test_models.py

```diff
@@ -42,6 +151,317 @@ def __init__(self, task) -> None:
+        end_train_time = perf_counter()
+        logger.debug(
+            f"Successfully trained the model in {end_train_time - start_train_time:.2f}s"
+        )
+        predictions_tmp = _get_probability_and_select_pred_columns(
+            test_data, model, post_transformer, id_a, id_b, dep_var
+        )
+        predict_train_tmp = _get_probability_and_select_pred_columns(
+            training_data, model, post_transformer, id_a, id_b, dep_var
+        )
+
+        test_pred = predictions_tmp.toPandas()
+        precision, recall, thresholds_raw = precision_recall_curve(
+            test_pred[f"{dep_var}"],
+            test_pred["probability"].round(2),
+            pos_label=1,
+        )
+        pr_auc = auc(recall, precision)
+        return pr_auc
+
+    # Returns a PR AUC list computation for inner training data on the given model
+    def _collect_inner_kfold_cv(
+        self,
+        inner_folds: list[pyspark.sql.DataFrame],
+        model_type: str,
+        hyperparams: dict[str, Any],
+        dep_var: str,
+        id_a: str,
+        id_b: str,
+    ) -> list[float]:
+        start_time = perf_counter()
+        # Collect auc values so we can pull out the highest
+        validation_results = []
+        for validation_index in range(len(inner_folds)):
+            validation_data = inner_folds[validation_index]
+            c_start_time = perf_counter()
+            training_data = self._combine_folds(inner_folds, ignore=validation_index)
+            c_end_time = perf_counter()
+            logger.debug(
+                f"Combined inner folds to make training data, except {validation_index}, took {c_end_time - c_start_time:.2f}"
+            )
+
+            cached_training_data = training_data.cache()
+            cached_validation_data = validation_data.cache()
+
+            # PRAUC = Precision Recall under the curve
+            prauc = self._train_model(
+                cached_training_data,
+                cached_validation_data,
+                model_type,
+                hyperparams,
+                dep_var,
+                id_a,
+                id_b,
+            )
+            training_data.unpersist()
+            validation_data.unpersist()
+            validation_results.append(prauc)
+        end_time = perf_counter()
+        logger.debug(
+            f"Inner folds: Evaluated model + params on {len(inner_folds)} folds in {end_time - start_time:.2f}"
+        )
+        logger.debug(f"Validation results {validation_results}")
+        return validation_results
+
+    # Returns a list of  ModelEval instances.
+    # This connects a score to each hyper-parameter combination. and the thresholds  listed with it in the config.
+    def _evaluate_hyperparam_combinations(
+        self,
+        all_model_parameter_combos,
+        inner_folds: list[pyspark.sql.DataFrame],
+        dep_var: str,
+        id_a: str,
+        id_b: str,
+        config,
+        training_conf,
```

Comment:
These names are pretty confusing. Maybe we can rename them to make it clear that one is the config dictionary and one is the name of the training config section. Or maybe we could just pass the training part of the dictionary to this function.

Reply:
I did both; consolidate the passing of config + training_conf (name) to the functions and instead pull out the training dict and calling it "training_settings". This simplifies accessing the training settings in places to be a bit cleaner and the names are more understandable. Really we need to remove the reliance on the config structure all over the place but that's a bigger change.